### PR TITLE
[fix](mtmv)use isManagedTable instead of check table type

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

The TableType of olap table is `OLAP`, and the TableType of materialized view is `MATERIALIZED_VIEW`. Many functions of olap are required for materialized views, but many previous functions filtered out materialized views by determining whether the TableType is `OLAP` and need to be changed to using `TableIf.isManagedTable()` to determine

not allow recycle mv
not allow recycle partition in mv
will ignore mv when recycle db

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java [hunk 1]
```diff
@@ -808,7 +808,7 @@ private void checkAndPrepareMeta() {
 
             // generate create replica task for all restored tables
             for (Table restoreTbl : restoredTbls) {
-                if (restoreTbl.getType() == TableType.OLAP) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java:811]: so async mv support backup and restore?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java [hunk 2]
```diff
@@ -112,7 +112,7 @@ public synchronized boolean allTabletsInRecycledStatus(List<Long> backendTabletI
     }
 
     private void addRecycledTabletsForTable(Set<Long> recycledTabletSet, Table table) {
-        if (table.getType() == TableType.OLAP) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java:115]: could we restore async mv from recycle bin? or we just mv it into recycle bin when we drop it?

**Review by Jibing-Li — DISMISSED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run feut

**Comment by zddr** [comment]
run cloud_p1

**Comment by zddr** [comment]
run buildall

**Review by Jibing-Li — APPROVED** [review]

**Pull request merged**